mlir/lib/Dialect/Vector/IR/VectorOps.cpp

```diff
@@ -2159,13 +2159,11 @@ class ExtractOpFromBroadcast final : public OpRewritePattern<ExtractOp> {
     // folding patterns.
     if (extractResultRank < broadcastSrcRank)
       return failure();
+    // For scalar result, the input can only be a zero-dim vector, which will
+    // be handled by the folder.
```

Comment:
"zero-dim" is IMO ambiguous and we should avoid that term. Note:
* `vector<0xf32>` - zero-dim vector
* `vector<f32>` - rank-zero vector

I know that in many places we use the term `0D`, but that usually means "zero dimensional` (and e.g. "2D" means two-dimensional). My suggest:
> For scalar results, let the fold pattern/method handle it.

Reply:
Replaced zero-dim with rank-0 vector. I like having the reasoning here, in case someone reads it and thinks why we decided to do this.